msprime/demography.py

```diff
@@ -3499,15 +3487,10 @@ def _matrix_exponential_poisson(A, n=5):
             Pk = (t / k) * np.matmul(P, Pk)
             expA = expA + Pk
         expA *= np.exp(-t)
-        for _ in range(nscales):
-            expA = np.matmul(expA, expA)
+        expA = np.linalg.matrix_power(expA, 2 ** nscales)
```

Comment:
I'm a bit nervous about this, not knowing about the numerical stability and efficiency of `np.linalg.matrix_power`? Oh, never mind! From [the docs](https://numpy.org/doc/stable/reference/generated/numpy.linalg.matrix_power.html) I see that
```
For positive integers n, the power is computed by repeated matrix squarings and matrix multiplications
```

Reply:
This is potentially not as obvious (one has to read the docs). But it's measurably faster when calling `mean_coalescence_time()`.